lib/logflare/sources.ex

```diff
@@ -40,31 +41,50 @@ defmodule Logflare.Sources do
 
   @spec create_source(map(), User.t()) :: {:ok, Source.t()} | {:error, Ecto.Changeset.t()}
   def create_source(source_params, user) do
+    postgres_backend_adapter_url =
+      Application.get_env(:logflare, :postgres_backend_adapter, []) |> Keyword.get(:url)
+
+    source_params =
+      source_params
+      |> Enum.map(fn {k, v} -> {to_string(k), v} end)
+      |> Map.new()
+      |> then(&Map.put(&1, "v2_pipeline", !!postgres_backend_adapter_url))
+
     source =
       user
       |> Ecto.build_assoc(:sources)
       |> Source.update_by_user_changeset(source_params)
       |> Repo.insert()
 
-    case source do
-      {:ok, source} ->
-        init_schema = SchemaBuilder.initial_table_schema()
+    case postgres_backend_adapter_url do
```

Comment:
should also factor in the single tenant mode flag.

Reply:
👍 makes sense